cmds/rm/rm_test.go

```diff
@@ -7,114 +7,246 @@
+				err:   nilerr,
+				stdin: no,
+			},
 		}
+	)
+
+	for _, tc := range testCases {
+
+		t.Run(tc.name, func(t *testing.T) {
+			d, err := ioutil.TempDir(os.TempDir(), "u-root.cmds.rm")
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer os.RemoveAll(d)
+
+			for _, f := range tmpFiles {
+				var (
+					err      error
+					filepath = path.Join(d, f.name)
+				)
+				if f.isdir {
+					err = os.Mkdir(filepath, f.mode)
+				} else {
+					err = ioutil.WriteFile(filepath, fbody, f.mode)
+				}
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+			input = bufio.NewReader(tc.stdin)
+			testRemove(t, d, tc)
+		})
 	}
-
-	return d, nil
 }
 
-// without any flag
-func Test_rm_1(t *testing.T) {
-	d, err := setup()
-	if err != nil {
-		t.Fatal("Error on setup of the test: creating files and folders.")
+func testRemove(t *testing.T, dir string, tc rmTestCase) {
+	var files = make([]string, len(tc.files))
+	for i, f := range tc.files {
+		files[i] = path.Join(dir, f.name)
 	}
-	defer os.RemoveAll(d)
-
-	fmt.Println("== Deleting files and empty folders (no args) ...")
-	files := []string{filepath.Join(d, "hi1.txt"), filepath.Join(d, "hi2.txt"), filepath.Join(d, "go.txt")}
 
 	flags.v = true
-	if err := rm(files); err != nil {
+	flags.r = tc.r
+	flags.f = tc.f
+	flags.i = tc.i
+
+	if err := rm(files); !tc.err(err) {
 		t.Error(err)
 	}
-}
 
-// using r flag
-func Test_rm_2(t *testing.T) {
-	d, err := setup()
-	if err != nil {
-		t.Fatal("Error on setup of the test: creating files and folders.")
+	if flags.i && tc.stdin.pos == 0 {
+		t.Error("Expected reading from stdin")
+	} else if !flags.i && tc.stdin.pos > 0 {
+		t.Errorf("Did not expect reading %d times from stdin", tc.stdin.pos)
+	}
+	if tc.stdin.overflowed {
+		t.Error("Read from stdin too many times")
 	}
-	defer os.RemoveAll(d)
 
-	flags.v = true
-	flags.r = true
-	fmt.Println("== Deleting folders recursively (using -r flag) ...")
-	files := []string{d}
-	if err := rm(files); err != nil {
-		t.Error(err)
+	for i, f := range tc.files {
+		_, err := os.Stat(path.Join(dir, f.name))
+		if tc.files[i].delete != os.IsNotExist(err) {
+			t.Errorf("File %q deleted: %t, expected: %t",
+				f.name, os.IsNotExist(err), tc.files[i].delete)
+		}
 	}
 }
 
-func TestForceRemove(t *testing.T) {
-	d, err := setup()
-	if err != nil {
-		t.Fatal("Error on setup of the test: creating files and folders.")
-	}
-	defer os.RemoveAll(d)
+// fakeStdin implements io.Reader and returns predfined list of answers,
+// suitable for stdin.
+type fakeStdin struct {
+	answers    []string
+	pos        int
+	overflowed bool
+}
 
-	flags.v = true
-	flags.r = false
-	flags.f = true
-	fmt.Println("== Deleting existing and nonexistent files (using -f flag) ...")
-	files := []string{
-		filepath.Join(d, "hi1.doc"), // does not exist
-		filepath.Join(d, "hi2.txt"), // exists
-		filepath.Join(d, "go.doc"),  // does not exist
-	}
-	if err := rm(files); err != nil {
-		t.Error(err)
+func newFakeStdin(answers ...string) *fakeStdin {
```

Comment:
And if you make the io.Reader change, this could actually possibly just be a 

```go
func newFakeStdin(answers ...string) io.Reader {
  return buffer.NewBufferString(strings.Join(answers, "\n"))
}
```

Reply:
Yes, but this will not allow me to check how the standard input has been used with given flags.